applications/dot_marketplace-phase2.md

```diff
@@ -0,0 +1,218 @@
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Migration from frame v1 to frame v2 | The [existing application backend](https://github.com/WowLabz/tasking_backend/blob/Phase1_Milestone3/pallets/pallet-tasking/src/lib.rs) will be migrated to frame v2 to account for the new features provided by the framework |
+| 2 | Chat Pallet | Chat functionality is to be exposed and consumed between the customer and the service provider to ease communication and this will be integrated with the tasking pallet's frame v2 version |
+|  |  |  |
+
+### **Milestone 2**
+
+* **Estimated duration:** 2 weeks
+* **FTE:**  2
+* **PTE:**  2
+* **Costs:**  15,900 USD  
+
+
+In continuation to previous work, this milestone involves creation of a on chain decentralized court to handle dispute resolution. Being a juror is one of the user incentives that can be achieved thanks to the rating module mentioned in the first phase of dot marketplace. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Decentralized Court Module | An on chain decentralized court for dispute resolution within the ecosystem. |
+| 1a | Disapprove Task  | In the case of a customer not being satisfied by the work submitted by the service provider (worker). A set of jurors is formed (court-summon) to resolve the dispute and pass a verdict. |
+| 1b | Disapprove Rating | The customer or the service provider, once they have received their rating for a particular task and are not satisfied by it. |
+| 1c | General Dispute | A general dispute function for cases that do not fall under the categories mentioned in 1a and 1b. |
+| 2 | Jury | The chain specification of the testnet is modified to include more users with necessary specifications to be a part of the jury. The specifications include having average user rating above a certain threshold and being an expert in the field of the task. A list of potential jurors are notified and they have a period of one day to accept jury duty, with the maximum number of juors capped to 5 per dispute. |
+| 3 | Voting module | Each juror can review the dispute and cast their vote which also includes their rating for both the customer and the worker. After a period of two days all the juror votes are counted and a winner is identified. |
```

Comment:
Here you say 2d, in the diagram it's 3d. Please clarify which one is the correct value and update accordingly.

Reply:
Ideally, you'd even make it configurable.

Reply:
3 days is the total time period for the entire court dispute. After the dispute is registered. 1 day is allotted for only accepting jury duty and after 1 day, the voting period starts which lasts for 2 days. And after 2 days the court is adjourned. And in those 2 days only the people who have accepted jury will be allowed to cast their votes.
The days can be configured based on how complex the task is. So we can provide a functionality before raising dispute, that for how many days does the raising party wants the trial to run.
So yes, it can be configured.